aten/src/ATen/autocast_mode.h

```diff
@@ -44,13 +48,35 @@ bool is_autocast_eligible(const Tensor& tensor, DeviceType device_type) {
       return tensor.is_xpu() && tensor.is_floating_point();
     case DeviceType::HPU:
       return tensor.is_hpu() && tensor.is_floating_point();
+    case DeviceType::XLA:
+      return tensor.is_xla() && tensor.is_floating_point();
     case DeviceType::PrivateUse1:
       return tensor.device().type() == DeviceType::PrivateUse1 &&
           tensor.is_floating_point();
     default:
       return false;
   }
 }
+
+// Policies correspond to op categories that need code-divergent handling.
+// Wrapper templates below are specialized based on a policy template parameter.
+enum class CastPolicy : uint8_t {
```

Comment:
Why was this moved to a (public) header file?

Reply:
I moved this into the public header file, so we could import the same CastPolicy in Pytorch/XLA. Alternatively, we could just copy it PyTorch/XLA repo to minimize changes here.

Reply:
ah ok seems fine :) Just wanted to make sure it was intentional.